### ISSUE TITLE
Diagnotistics for multi-node setup

### ISSUE THREAD

**Issue by seberg** [nv-legate/legate.core#631]
Setting up multi-node support, it may be easy to forget the correct gasnet/UCX setup and end up with very slow connectivity.  That could be a `legate --info` or so printing out informations about what capabilities legate was compiled with.

In the specific case, we had hanging behavior, which had to do with the `opempi` setup (although very simple `mpirun` calls succeeded).

An alternative/addition might be to add some documentation on how to verify profiling (maybe the cunumeric examples could be used for that even).

**Comment by manopapad** [comment]
> That could be a legate --info or so printing out informations about what capabilities legate was compiled with.

What kind of information would you want to see? I'd expect CUDA/OpenMP/networking support would be the top items here. @bryevdv how feasible would it be to query this kind of information from the driver?

> In the specific case, we had hanging behavior, which had to do with the opempi setup (although very simple mpirun calls succeeded).
>
> An alternative/addition might be to add some documentation on how to verify profiling (maybe the cunumeric examples could be used for that even).

Yes, documentation of multi-node setup is in the plans for the next release. We will mention some common pitfalls, but TBH there are so many pitfalls when using GASNet/UCX/OpenMPI (especially when combined with conda) that we can't hope to cover everything.

**Comment by seberg** [comment]
Yeah, OpenMP/networking mainly I think, CUDA would be nice also.  I.e. I have set up legate with UCX, but am not sure I can get diagnostics telling me that my two node setup uses infiniband via UCX?

Another thing is that we experienced a hang due to `mpirun`.  While annoying, I am not sure that is something legate can really help with (besides maybe linking to mpi docs on how to troubleshoot).

**Comment by bryevdv** [comment]
What kind of information would you want to see? I'd expect CUDA/OpenMP/networking support would be the top items here. @bryevdv<https://github.com/bryevdv> how feasible would it be to query this kind of information from the driver?
This seems like it would be fairly straightforward in principle. Some of the info might need to come from the legate/realm query API? I am not sure what the current status of that is. if hI'd suggest making an issue to discuss specific details/requirements.

Bryan
________________________________

**Comment by manopapad** [comment]
> I have set up legate with UCX, but am not sure I can get diagnostics telling me that my two node setup uses infiniband via UCX?

For this particular scenario we can output the information from our side that Realm was compiled against UCX, but you'd then need to go ask UCX if it was compiled with Infiniband support.

> I'd suggest making an issue to discuss specific details/requirements.

Opened #632
